### ISSUE TITLE
Thermostat operating states

### ISSUE THREAD

**Issue by Ittiz** [domoticz/domoticz#2170]
I'm not sure if the is an open-z-wave issue or a Domoticz issue.  I'm inclined to think a Domoticz issue since there seems to be code to get the values in open-z-wave but "THERMOSTAT OPERATING STATE" and "FAN STATE" both are unhandled in Domoticz.  I found I could bypass the issue using an LUA HTTP poller to directly get the info from open z-wave, which doesn't update the states unless it requests an update from the thermostat, usually done by polling the thermostat.  However I prefer to avoid z-wave polling.  So maybe it's a problem with both I don't know, but it would be extremely nice if the z-wave thermostat had "THERMOSTAT OPERATING STATE" and "FAN STATE" devices in Domoticz so I would easily able to see if the furnace is on or not.

**Comment by gizmocuz** [comment]
Currently implemented:

COMMAND_CLASS_THERMOSTAT_SETPOINT
COMMAND_CLASS_THERMOSTAT_MODE
COMMAND_CLASS_THERMOSTAT_FAN_MODE
ZDTYPE_SENSOR_THERMOSTAT_CLOCK

There is no thermostat state class, but maybe you meant 'mode' ?

Some sensors need polling, which is quite normal, you can select the node, and enable polling in the zwave setup screen

**Comment by Ittiz** [comment]
Not mode, state.  My thermostat does have modes like you mentioned, which allows you to set whether it's heating, cooling or auto, but it also has states as well. I looked around and found you had a discussion with someone about this thermostat and issue in the past on the [Domoticz forum](http://www.domoticz.com/forum/viewtopic.php?f=24&t=3672&p=172401l).  However the THERMOSTAT OPERATING STATE (command class 0x42) shows you what the thermostat is trying to do right now.   The other thing is the THERMOSTAT FAN STATE (command class 0x45) which tells you if the central air fan is idle or running.  Oh I also noticed another unhandled class called INDICATOR (command class 0x87).  Anyway I found [the Sigma designs doc](http://z-wave.sigmadesigns.com/wp-content/uploads/2016/08/SDS12652-13-Z-Wave-Command-Class-Specification-N-Z.pdf) that describes these command classes in detail.  THERMOSTAT OPERATING STATE is pages 296 to 302 and THERMOSTAT FAN STATE is pages 283 and 284.  I found command class INDICATOR was defined in [this doc](http://zwavepublic.com/sites/default/files/command_class_specs_2017A/SDS13781-5%20Z-Wave%20Application%20Command%20Class%20Specification.pdf) on pages 190 to 201.

**Comment by gizmocuz** [comment]
I only know the indicator is used to specify a color on some sensors... it's pretty static data, thats why i did not implement it.
If you could help implementing the themostat/fan state that would be great as i have no such devices

**Comment by Ittiz** [comment]
Let me know what you need me to do.

**Comment by gizmocuz** [comment]
You could add/test the needed code in the openzwave class

**Comment by Ittiz** [comment]
I could test it if you could compile me an exe for windows.  I don't have a compiler.

**Comment by gizmocuz** [comment]
Sorry, i was under the impression that you could code. This because you are writing this on github and not on the forum.
You can download visual studio for free from Microsoft (Community edition), but it looks like another developer with this hardware has to implement this

**Comment by Ittiz** [comment]
I did post this exact issue [on the forum](http://www.domoticz.com/forum/viewtopic.php?f=31&t=21129&p=163646#p163646) a month or two ago and got no replies.  My big issue is I don't have the time these days, which is why I haven't bothered setting up a testing environment for a compiler (I only code in VMs).  Although while I was writing this I remembered I have a HD laying around somewhere with a visual studio coding VM on it.  It's Win7 but I don't think that'll be an issue.  If I can find the HD and the time I'll take a look at the code, but I can't commit to anything.
